### PR TITLE
Eng 18234 - Speed up the final kit builds

### PR DESCRIPTION
**Pull request by rmorgenstein**
Some tweaks to the kit builds.
1. Stop making the "Pro" kit. This saves space more than time.
2. Speed up cloning: On OSX only clone repo.  Do shallow clone of exact branch, if possible
3. On OSX only build ee, not full community distro

### CODE REVIEW

File: tools/kit_tools/build_kits.py [hunk 1]
```diff
@@ -16,6 +16,35 @@
...
+        #Maybe it was a sha and needs a full clone
+        try:
+            #Check if it is hex
+            int(treeish, 16)
+        except ValueError:
+            return False
+        #Okay, it's hex, so lets try a real clone + checkout
+        run("git clone -q %s" % repo)
+        directory = repo.split("/")[-1]
+        result = run("cd %s; git checkout %s" % directory, treeish , warn_only=True)
```
Comment (dave-voltdb) [inline tools/kit_tools/build_kits.py]: I think you need   (directory, treeish)   

```
>>> print "foo %s %s" % "a","b"
TypeError: not enough arguments for format string
```

File: tools/kit_tools/build_kits.py [hunk 2]
```diff
@@ -31,22 +60,23 @@ def checkoutCode(voltdbGit, proGit, rbmqExportGit, gitloc):
         # do the checkouts, collect checkout errors on both community &
         # pro repos so user gets status on both checkouts
         message = ""
-        run("git clone -q %s/voltdb.git" % gitloc)
-        result = run("cd voltdb; git checkout %s" % voltdbGit, warn_only=True)
-        if result.failed:
-            message = "VoltDB checkout failed. Missing branch %s." % rbmqExportGit
```
Comment (dave-voltdb) [inline tools/kit_tools/build_kits.py]: I know this is now dead code (!) but why was rbmqExportGit used in the error message?

The same issue was cut and pasted into the update-cataloglib script.

Comment (rmorgenstein) [reply tools/kit_tools/build_kits.py]: I'm revisiting that script at some point and will look at it then.

File: tools/kit_tools/build_kits.py (hunk 1 above)
Comment (rmorgenstein) [reply tools/kit_tools/build_kits.py]: you are correct. I had to fix some other logic as well.

**Pull request merged**